lambda/src/main/scala/pricemigrationengine/handlers/NotificationHandler.scala

```diff
@@ -14,45 +14,76 @@ object NotificationHandler extends CohortHandler {
 
   val Successful = 1
   val Unsuccessful = 0
-
   val Cancelled_Status = "Cancelled"
 
-  private val NotificationLeadTimeDays = 49
+  // We are starting the notification process for any item whose start date is less than 49 (StandardNotificationLeadTime)
+  // days away, but because it can happen, we also need to react to items which for *any* reason have
+  // not been processed on time (meaning early enough) and for which the previously computed startDate is
+  // too close (less than MinNotificationLeadTime away) from now and would not give enough time to the letters.
+
+  private val StandardNotificationLeadTime = 49
+  private val MinNotificationLeadTime = 35
 
   def main(
       brazeCampaignName: String
   ): ZIO[Logging with CohortTable with SalesforceClient with EmailSender, Failure, HandlerOutput] = {
     for {
       today <- Clock.currentDateTime.map(_.toLocalDate)
       count <- CohortTable
-        .fetch(SalesforcePriceRiceCreationComplete, Some(today.plusDays(NotificationLeadTimeDays)))
-        .mapZIO(sendNotification(brazeCampaignName))
+        .fetch(SalesforcePriceRiceCreationComplete, Some(today.plusDays(StandardNotificationLeadTime)))
+        .mapZIO(item => sendNotification(brazeCampaignName)(item, today))
         .runFold(0) { (sum, count) => sum + count }
       _ <- Logging.info(s"Successfully sent $count price rise notifications")
     } yield HandlerOutput(isComplete = true)
   }
 
+  def thereIsEnoughNotificationLeadTime(today: LocalDate, cohortItem: CohortItem): Boolean = {
+    // To help with backward compatibility with existing tests, we apply this condition from 1st Dec 2022.
+    if (today.isBefore(LocalDate.of(2022, 12, 1))) {
```

Comment:
```suggestion
    if (today.isBefore(LocalDate.of(2020, 12, 1))) {
```

Reply:
Doesn't really matter but the tests are using 2020 so maybe makes it a bit more straightforward if someone new reads this.